### PR TITLE
Update Vintage Story egg for prerelease support

### PR DESCRIPTION
**Pull request by Kamisoi**
# Description

Update of Vintage Story egg to support unstable and pre release branches, mainly due to [v1.18](https://info.vintagestory.at/v1dot18#pre5) prereleases being available (sadly, their API endpoints do not yet have that updated, as seen under respective URLs for API endpoints)
Change of README to reflect accessibility of unstable and pre file branches.

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [x] You verify that the start command applied does not use a shell script
  * [x] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/vintage_story/egg-vintage-story.json [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
-    "name": "Vintage Story",
-    "author": "mail@wuffy.eu",
+    "exported_at": "2023-03-08T21:10:40+00:00",
+    "name": "Vintage Story - Custom",
+    "author": "kamisoi@meowstard.cat",
     "description": "Vintage Story is an uncompromising wilderness survival sandbox game inspired by lovecraftian horror themes. Find yourself in a ruined world reclaimed by nature and permeated by unnerving temporal disturbances. Relive the advent of human civilization, or take your own path.",
-    "image": "quay.io\/parkervcp\/pterodactyl-images:debian_mono-6-complete",
+    "features": null,
+    "docker_images": {
+        "quay.io\/parkervcp\/pterodactyl-images:debian_mono-6-complete": "quay.io\/parkervcp\/pterodactyl-images:debian_mono-6-complete"
```
Comment (QuintenQVD0) [inline game_eggs/vintage_story/egg-vintage-story.json:13]: Please change this to the update image: `ghcr.io/parkervcp/yolks:mono_latest`

File: game_eggs/vintage_story/egg-vintage-story.json [hunk 2]
```diff
@@ -17,38 +22,51 @@
...
...son\"\r\n            ;;\r\n        *)\r\n            echo \"Invalid stage; fallback to stable\"\r\n            API_URL=\"http:\/\/api.vintagestory.at\/stable.json\"\r\n            ;;\r\n    esac\r\n    DOWNLOAD_URL=`curl -s \"$API_URL\" | jq -r 'if ([.[]] | .[0].server.urls.cdn) != null then [.[]] | .[0].server.urls.cdn else [.[]] | .[0].server.urls.local end'`;\r\n# Use specified server release\r\nelse\r\n    case \"${BRANCH}\" in\r\n    \"stable\")\r\n        DOWNLOAD_URL=https:\/\/cdn.vintagestory.at\/gamefiles\/stable\/vs_server_${RELEASE_VERSION}.tar.gz\r\n        ;;\r\n    \"unstable\")\r\n        DOWNLOAD_URL=https:\/\/account.vintagestory.at\/files\/unstable\/vs_server_${RELEASE_VERSION}.tar.gz\r\n        ;;\r\n    \"pre\")\r\n        DOWNLOAD_URL=https:\/\/cdn.vintagestory.at\/gamefiles\/pre\/vs_server_${RELEASE_VERSION}.tar.gz\r\n        ;;\r\n    *)\r\n        echo \"Invalid stage; fallback to stable\"\r\n        DOWNLOAD_URL=https:\/\/cdn.vintagestory.at\/gamefiles\/stable\/vs_server_${RELEASE_VERSION}.tar.gz\r\n        ;;\r\n    esac\r\nfi\r\n\r\ncd \/mnt\/server\/\r\n\r\ncurl -o vs_server.tar.gz ${DOWNLOAD_URL}\r\ntar -xzf vs_server.tar.gz\r\n\r\nrm vs_server.tar.gz",
```
Comment (QuintenQVD0) [inline game_eggs/vintage_story/egg-vintage-story.json:25]: You dont set debian to non interactive here? And you should not need screen

File: game_eggs/vintage_story/egg-vintage-story.json [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
     "meta": {
-        "version": "PTDL_v1"
+        "version": "PTDL_v2",
+        "update_url": null
     },
-    "exported_at": "2020-03-20T23:54:00+01:00",
-    "name": "Vintage Story",
-    "author": "mail@wuffy.eu",
+    "exported_at": "2023-03-08T21:10:40+00:00",
+    "name": "Vintage Story - Custom",
```
Comment (QuintenQVD0) [inline game_eggs/vintage_story/egg-vintage-story.json:8]: You can not change the email nore the name of the egg

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/vintage_story/egg-vintage-story.json (hunk 3 above)
Comment (Kamisoi) [reply game_eggs/vintage_story/egg-vintage-story.json:8]: Forgot to change that one back; was testing on the side and email was forcefully changed

File: game_eggs/vintage_story/egg-vintage-story.json (hunk 2 above)
Comment (Kamisoi) [reply game_eggs/vintage_story/egg-vintage-story.json:25]: Screen was previously in the egg, due to [official install instructions](https://wiki.vintagestory.at/index.php?title=Setting_up_a_Multiplayer_Server#Requirements_for_Ubuntu_(Debian,_Mint,_...)); will remove here.

Comment (Kamisoi) [reply game_eggs/vintage_story/egg-vintage-story.json:25]: `DEBIAN_FRONTEND=noninteractive` was purely added to suppress few errors, eg: `debconf: unable to initialize frontend`; mistakenly left it at `apt update` command :/

**Comment by QuintenQVD0** [comment]
Please try setting the install script to this: 
and then test if it still works

```bash
#!/bin/bash
# Vintage Story Installation Script
#
# Server Files: /mnt/server
apt update
apt -y install wget curl jq

BRANCH="${FILES_BRANCH}"

# Fetching latest stable server release
if [ -z "${RELEASE_VERSION}" ] || [ "${RELEASE_VERSION}" == "latest" ]; then
 case "${BRANCH}" in
 "stable")
 API_URL="http://api.vintagestory.at/stable.json"
 ;;
 "unstable")
 API_URL="http://api.vintagestory.at/unstable.json"
 ;;
 "pre")
 API_URL="http://api.vintagestory.at/pre.json"
 ;;
 *)
 echo "Invalid stage; fallback to stable"
 API_URL="http://api.vintagestory.at/stable.json"
 ;;
 esac
 DOWNLOAD_URL=$(curl -SsL "$API_URL" | jq -r 'if ([.[]] | .[0].server.urls.cdn) != null then [.[]] | .[0].server.urls.cdn else [.[]] | .[0].server.urls.local end');
# Use specified server release
else
 case "${BRANCH}" in
 "stable")
 DOWNLOAD_URL=https://cdn.vintagestory.at/gamefiles/stable/vs_server_${RELEASE_VERSION}.tar.gz
 ;;
 "unstable")
 DOWNLOAD_URL=https://account.vintagestory.at/files/unstable/vs_server_${RELEASE_VERSION}.tar.gz
 ;;
 "pre")
 DOWNLOAD_URL=https://cdn.vintagestory.at/gamefiles/pre/vs_server_${RELEASE_VERSION}.tar.gz
 ;;
 *)
 echo "Invalid stage; fallback to stable"
 DOWNLOAD_URL=https://cdn.vintagestory.at/gamefiles/stable/vs_server_${RELEASE_VERSION}.tar.gz
 ;;
 esac
fi

cd /mnt/server/ || exit

curl -o vs_server.tar.gz "${DOWNLOAD_URL}"
tar -xzf vs_server.tar.gz

rm vs_server.tar.gz

echo "-----------------------------------------"
echo "Installation completed..."
echo "-----------------------------------------"
```

**Comment by Kamisoi** [comment]
> Please try setting the install script to this: and then test if it still works

Works fine, although there are still logs than can be easily mitigated by keeping DEBIAN_NONINTERACTIVE flag for package installation part
```syslog
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.28.1 /usr/local/share/perl/5.28.1 /usr/lib/x86_64-linux-gnu/perl5/5.28 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.28 /usr/share/perl/5.28 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
```

**Comment by Kamisoi** [comment]
Removed DEBIAN_NONINTERACTIVE alltogether, hopefully now is all good ^^

**Review by QuintenQVD0 — APPROVED** [review]
LGTM! thank you

**Comment by QuintenQVD0** [comment]
I just cleaned up some left over that is not more used as of panel v1 and update the install image to debian 11

**Pull request merged**